src/workspaces/list/NoContentMessage.test.tsx

```diff
@@ -0,0 +1,67 @@
+import { screen } from '@testing-library/react';
+import React from 'react';
+import { asMockedFn, renderWithAppContexts as render } from 'src/testing/test-utils';
+import { defaultGoogleWorkspace } from 'src/testing/workspace-fixtures';
+import { NoWorkspacesMessage } from 'src/workspaces/common/NoWorkspacesMessage';
+import { CategorizedWorkspaces } from 'src/workspaces/list/CategorizedWorkspaces';
+import { NoContentMessage } from 'src/workspaces/list/NoContentMessage';
+import { WorkspaceFilterValues } from 'src/workspaces/list/WorkspaceFilters';
+
+jest.mock('src/workspaces/common/NoWorkspacesMessage');
+
+describe('NoContentMessage', () => {
+  const workspaces: CategorizedWorkspaces = {
+    myWorkspaces: [defaultGoogleWorkspace],
+    newAndInteresting: [],
+    featured: [],
+    public: [],
+  };
+
+  const emptyWorkspaces: CategorizedWorkspaces = {
+    myWorkspaces: [],
+    newAndInteresting: [],
+    featured: [],
+    public: [],
+  };
+
+  const filters: WorkspaceFilterValues = {
+    keywordFilter: '',
+    accessLevels: [],
+    tab: 'myWorkspaces',
+    tags: [],
+  };
+
+  it('displays loading message if workspaces are loading', () => {
+    // Arrange
+    const props = { loadingWorkspaces: true, workspaces, filters };
+
+    // Act
+    render(<NoContentMessage {...props} />);
+
+    // Assert
+    expect(screen.getByText('Loading...')).toBeInTheDocument();
+  });
+
+  it('displays no workspaces message if there are no workspaces', () => {
+    // Arrange
+    const props = { loadingWorkspaces: false, workspaces: emptyWorkspaces, filters };
+    asMockedFn(NoWorkspacesMessage).mockReturnValue(<div>mockNoWorkspacesMessage</div>);
+
+    // Act
+    render(<NoContentMessage {...props} />);
+
+    // Assert
+    expect(screen.getByText('mockNoWorkspacesMessage')).toBeInTheDocument();
+  });
+
+  it('displays no matching workspaces message otherwise', () => {
+    // Arrange
+    const props = { loadingWorkspaces: false, workspaces, filters };
+
+    // Act
+    render(<NoContentMessage {...props} />);
+
+    // Assert
+    expect(screen.getByText('No matching workspaces')).toBeInTheDocument();
```

Comment:
This test is confusing because in this case, filters is set to be `myWorkspaces`, and the `workspaces` variable contains a workspace in the `myWorkspaces` array. The reality is that we wouldn't call `NoContentMessage` in this case.

I would recommend making the test case inputs more sensical, like the filter being `newAndInteresting`, which has no workspaces in your test data.

Reply:
I switched to two different tests to test more cases which should result in the "no matching workspaces" message. Both of them now fulfill this criteria for realism--one filters on `myWorkspaces` but also searches for a keyword which doesn't match the default workspace, and one filters on `featured`, which is empty.